pandas/core/internals/concat.py

```diff
@@ -31,6 +31,47 @@
     from pandas import Index
 
 
+def concatenate_array_managers(
+    mgrs_indexers, axes: List[Index], concat_axis: int, copy: bool
+) -> Manager:
+    """
+    Concatenate array managers into one.
+
+    Parameters
+    ----------
+    mgrs_indexers : list of (ArrayManager, {axis: indexer,...}) tuples
+    axes : list of Index
+    concat_axis : int
+    copy : bool
+
+    Returns
+    -------
+    ArrayManager
+    """
+    # reindex all arrays
+    mgrs = []
+    for mgr, indexers in mgrs_indexers:
+        for ax, indexer in indexers.items():
+            mgr = mgr.reindex_indexer(
+                axes[ax], indexer, axis=ax, allow_dups=True, do_integrity_check=False
+            )
+        mgrs.append(mgr)
+
+    # concatting along the rows -> concat the reindexed arrays
+    # TODO(ArrayManager) doesn't yet preserve the correct dtype
```

Comment:
does this comment apply to both cases or just concat_axis==1?  (this ambiguity is why i like to put comments inside below the "if"/"else")

Reply:
Moved the comment (yes, it applies only to the axis==1)